crates/cdk/src/mint/ln.rs

```diff
@@ -45,6 +45,20 @@ impl Mint {
 
         let mut tx = self.localstore.begin_transaction().await?;
 
+        if quote.payment_method == PaymentMethod::Bolt11 {
+            // reload the quote, as it state may have changed
+            *quote = tx
+                .get_mint_quote(&quote.id)
+                .await?
+                .ok_or(Error::UnknownQuote)?;
```

Comment:
I think we should do this for all payment methods? The same race condition could occur.